ratis-server-api/src/main/java/org/apache/ratis/statemachine/TransactionContext.java

```diff
@@ -96,9 +97,28 @@ public interface TransactionContext {
   /**
    * Returns the committed log entry
    * @return the committed log entry
+   *
+   * @deprecated use {@link TransactionContext#getLogEntryRef()} instead.
```

Comment:
Let's update the javadoc to mention "copy".
```java
   * @return a copy of the committed log entry if it exists; otherwise, returns null.
   *
   * @deprecated Use {@link #getLogEntryRef()} or {@link #getLogEntryUnsafe()} to avoid copying.
```